aptos-move/aptos-vm/src/aptos_vm.rs

```diff
@@ -848,6 +864,117 @@ impl AptosVM {
         Ok(())
     }
 
+    fn validate_and_execute_batched_calls(
+        &self,
+        resolver: &impl AptosMoveResolver,
+        session: &mut SessionExt,
+        gas_meter: &mut impl AptosGasMeter,
+        traversal_context: &mut TraversalContext,
+        senders: Vec<AccountAddress>,
+        batched_calls: &[BatchedFunctionCall],
+        txn_data: &TransactionMetadata,
+    ) -> Result<(), VMStatus> {
+        let mut batched_return_values = BatchedExecutionResult {
+            return_values: vec![],
+        };
+        for batched_call in batched_calls {
+            self.validate_and_execute_batched_call(
+                resolver,
+                session,
+                gas_meter,
+                traversal_context,
+                senders.clone(),
+                batched_call,
+                &mut batched_return_values,
+                txn_data,
+            )?;
+        }
+        for return_values in batched_return_values.return_values {
+            for value in return_values {
+                if !value.ability.has_drop() && !value.consumed {
+                    return Err(
+                        PartialVMError::new(StatusCode::INVALID_MAIN_FUNCTION_SIGNATURE)
+                            .finish(Location::Undefined)
+                            .into_vm_status(),
+                    );
+                }
+            }
+        }
+        Ok(())
+    }
+
+    fn validate_and_execute_batched_call(
+        &self,
+        _resolver: &impl AptosMoveResolver,
+        session: &mut SessionExt,
+        gas_meter: &mut impl AptosGasMeter,
+        traversal_context: &mut TraversalContext,
+        senders: Vec<AccountAddress>,
+        batched_call: &BatchedFunctionCall,
+        batched_return_values: &mut BatchedExecutionResult,
+        _txn_data: &TransactionMetadata,
+    ) -> Result<(), VMStatus> {
+        let module_id = traversal_context
+            .referenced_module_ids
+            .alloc(batched_call.module.clone());
+        session.check_dependencies_and_charge_gas(gas_meter, traversal_context, [(
+            module_id.address(),
+            module_id.name(),
+        )])?;
+
+        let (function, is_friend_or_private) = session.load_function_and_is_friend_or_private_def(
```

Comment:
I removed this function, you can get a function directly now and call `is_friend_or_private()`